flent/runners.py

```diff
@@ -1080,39 +1095,108 @@ def filter_np_parent(self, part):
             if None is f_ports:
                 raise ParseException()
 
-            dst_p = int(f_ports.group('dst_p').split(":")[-1])
+            dst_shards = f_ports.group('dst_p').split(":")
+            # cure by checking ip_version
+            dst_addr = ""
+            if len(dst_shards) > 2:
+                dst_addr = ":".join(dst_shards[:-1])
+            else:
+                dst_addr = ".".join(dst_shards[:-1])
 
-            if self.par_pid == pid_str and dst_p not in self.exclude_ports:
-                sub_part.append(sp)
+            dst_p = dst_shards[-1]
 
-        if 1 != len(sub_part):
-            raise ParseException()
+            if self.filt_pid == pid_str and dst_p not in self.exclude_ports:
+                sub_parts_res.append((sp, (dst_addr, dst_p)))
 
-        return sub_part[0]
+        self.post_filter_pred(sub_parts_res)
 
-    def parse_part(self, part):
-        sub_part = self.filter_np_parent(part)
+        return sub_parts_res
 
-        timestamp = self.time_re.search(part)
+    def post_filter_pred(self, sub_parts):
+        if self.ext_run:
+            pass
+        else:
+            if 1 != len(sub_parts):
+                raise ParseException()
+
+    def form_sub_raw_vals(self, cwnd, rtt, rtt_var, bbr, flow_spec):
+        bbr_keys = ['bbr_bw', 'bbr_mrtt', 'bbr_pacing_gain', 'bbr_cwnd_gain']
+        if self.ext_run:
+            sub_raw_val = {}
+            for k, val in zip(['cwnd', 'rtt', 'rtt_var'], [cwnd, rtt, rtt_var]):
```

Comment:
You can probably get rid of the repetition here in a similar way (named groups in regexes + groupdict())